src/frequenz/client/dispatch/_client.py

```diff
@@ -91,10 +104,18 @@ async def list(
             end_to: optional, filter by end_time < end_to.
             active: optional, filter by active status.
             dry_run: optional, filter by dry_run status.
+            page_size: optional, number of dispatches to return per page. Conflicts with page_token.
+            page_token: optional, token for the next page, will continue iterating from this token.
 
         Returns:
             An async iterator of dispatches.
+
+        Raises:
+            ValueError: If both page_size and page_token are provided.
         """
+        if page_size is not None and page_token is not None:
+            raise ValueError("page_size and page_token are mutually exclusive")
```

Comment:
You can also use an overload to detect misuses with `mypy` (I also strongly recommend making arguments keyword only with so many `datetime`/`int`/`bool`/`str`).

I think it should be something like this:
```py
    @overload
    async def list(
        self,
        microgrid_id: int,
        component_selectors: Iterator[ComponentSelector] = iter(()),
        *,
        start_from: datetime | None = None,
        start_to: datetime | None = None,
        end_from: datetime | None = None,
        end_to: datetime | None = None,
        active: bool | None = None,
        dry_run: bool | None = None,
        page_size: int | None = None,
        page_token: Literal[None],
    ) -> AsyncIterator[Dispatch]:

    async def list(
        self,
        microgrid_id: int,
        component_selectors: Iterator[ComponentSelector] = iter(()),
        *,
        start_from: datetime | None = None,
        start_to: datetime | None = None,
        end_from: datetime | None = None,
        end_to: datetime | None = None,
        active: bool | None = None,
        dry_run: bool | None = None,
        page_size: Literal[None],
        page_token: str | None = None,
    ) -> AsyncIterator[Dispatch]:

    async def list(
        self,
        microgrid_id: int,
        component_selectors: Iterator[ComponentSelector] = iter(()),
        *,
        start_from: datetime | None = None,
        start_to: datetime | None = None,
        end_from: datetime | None = None,
        end_to: datetime | None = None,
        active: bool | None = None,
        dry_run: bool | None = None,
        page_size: int | None = None,
        page_token: str | None = None,
    ) -> AsyncIterator[Dispatch]:
```